src/sql/platform/connection/common/connectionManagement.ts

```diff
@@ -57,7 +57,7 @@ export interface IConnectionResult {
 export interface IConnectionCallbacks {
 	onConnectStart(): void;
 	onConnectReject(error?: string): void;
-	onConnectSuccess(params?: INewConnectionParams): void;
+	onConnectSuccess(params: INewConnectionParams, profile: IConnectionProfile): void;
```

Comment:
The connection dialog can change the profile which means the connection management service and the caller are referring to different profiles. This allows the the caller to get the updated profile.